CorsixTH/Src/th_sound.cpp

```diff
@@ -248,8 +248,7 @@ void sound_player::populate_from(sound_archive* pArchive) {
   sounds = new Mix_Chunk*[pArchive->get_number_of_sounds()];
   for (; sound_count < pArchive->get_number_of_sounds(); ++sound_count) {
     sounds[sound_count] = nullptr;
-    SDL_RWops* pRwop = pArchive->load_sound(sound_count);
-    if (pRwop) {
+    if (SDL_RWops* pRwop = pArchive->load_sound(sound_count)) {
```

Comment:
Please don't have side effects in conditions.

The compiler will merge the assignment and the test.

Reply:
This is perfectly valid and encoraged c++ style. Instantiation in if is unambiguous and reduces scope.